contracts/metatx/ERC2771Forwarder.sol

```diff
@@ -186,31 +196,36 @@ contract ERC2771Forwarder is EIP712, Nonces {
      */
     function _validate(
         ForwardRequestData calldata request
-    ) internal view virtual returns (bool alive, bool signerMatch, address signer) {
-        signer = _recoverForwardRequestSigner(request);
-        return (request.deadline >= block.timestamp, signer == request.from, signer);
+    ) internal view virtual returns (bool unexpired, bool signerMatch, address signer) {
+        (address recovered, bool isValid) = _recoverForwardRequestSigner(request);
+        return (isValid && request.deadline >= block.timestamp, recovered == request.from, recovered);
```

Comment:
Signature validity should be reflected in the `signerMatch` return value.

```suggestion
        return (request.deadline >= block.timestamp, isValid && recovered == request.from, recovered);
```

Reply:
Fixed in 53b4feb